iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift

```diff
@@ -31,134 +31,183 @@ import SalesforceSDKCore
 struct NativeLoginView: View {
     @Environment(\.colorScheme) var colorScheme
     
+    /// The reCAPTCHA client used to obtain reCAPTCHA tokens when needed for Salesforce Headless Identity API requests.
+    @EnvironmentObject var reCaptchaClientObservable: ReCaptchaClientObservable
+    
+    /// The navigation path.
+    @EnvironmentObject var navigationPathObservable: NavigationPathObservable
```

Comment:
SwiftUI Noob At Work 🚧 This pull request introduces multi-level navigation via `NavigationStack` to the template app for the first time.  So, I followed some Apple docs and created a path of custom `NavigationDestination` enums to drive the flow in and out of the navigation stack with associated values.  It sure looks cool to my rookie eyes, but I'd be game to see how the team likes it.

I was curious to maybe _not_ use an observable for this one and see how the binding to the `NavigationStack` could be simplified.  This works really well though as-is.

Reply:
The `NavigationStack` seems a bit overkill to me when you could show either username/password or OTP flow with a simple if else instead.  It's not inherently bad, but it makes the template a bit more complex as an example.  Also, adding the nav stack appears to have broken the existing back button functionality for multiuser loginf.

Reply:
When I thought about how I'd have written this in Kotlin Compose, using conditional layout logic on a bound variable worked perfectly and was a lot simpler - just like you said.  Thanks for the nudge.

Reply:
That also fixed the back navigation, which I restored to the way it is on `dev` exactly.